JOSESwift/Sources/CryptoImplementation/RSA.swift

```diff
@@ -50,24 +50,42 @@ fileprivate extension AsymmetricKeyAlgorithm {
         switch self {
         case .RSA1_5:
             return .rsaEncryptionPKCS1
+        case .RSAES_OAEP:
+            return .rsaEncryptionOAEPSHA1
         }
     }
 
     /// Checks if the plain text length does not exceed the maximum
     /// for the chosen algorithm and the corresponding public key.
     func isPlainTextLengthSatisfied(_ plainText: Data, for publicKey: SecKey) -> Bool {
+        let mLen: Int = plainText.count
+        let k = SecKeyGetBlockSize(publicKey)
         switch self {
         case .RSA1_5:
             // For detailed information about the allowed plain text length for RSAES-PKCS1-v1_5,
-            // please refer to the RFC(https://tools.ietf.org/html/rfc3447#section-7.2).
-            return plainText.count <= (SecKeyGetBlockSize(publicKey) - 11)
+            // please refer to the RFC(https://tools.ietf.org/html/rfc3447#section-7.2.1).
+            return mLen <= (k - 11)
+        case .RSAES_OAEP:
+            // For detailed information about the allowed plain text length for RSAES_OAEP,
+            // please refer to the RFC(https://tools.ietf.org/html/rfc3447#section-7.1.1).
+            // hLen = input limitation for the hash function (2^61 - 1 octets for SHA-1)
+            let hLen = (1 << 61) - 1
+            return mLen <= (k - 2 * hLen - 2)
         }
     }
 
     func isCipherTextLenghtSatisfied(_ cipherText: Data, for privateKey: SecKey) -> Bool {
         switch self {
         case .RSA1_5:
+            // For detailed information about the allowed cipher length for RSAES-PKCS1-v1_5,
+            // please refer to the RFC(https://tools.ietf.org/html/rfc3447#section-7.2.2).
             return cipherText.count == SecKeyGetBlockSize(privateKey)
+        case .RSAES_OAEP:
+            // For detailed information about the allowed cipher length for RSAES_OAEP,
+            // please refer to the RFC(https://tools.ietf.org/html/rfc3447#section-7.1.2).
+            // hLen = input limitation for the hash function (2^61 - 1 octets for SHA-1)
+            let hLen = (1 << 61) - 1
```

Comment:
Same as above, SHA-1 [digest length is 20](https://tools.ietf.org/html/rfc3174#section-1).

Reply:
Fixed